app/utils/report/boot.py

```diff
@@ -14,6 +14,7 @@
 """Create the boot email report."""
 
 import itertools
+import jenkins
```

Comment:
Hmmm... now I see.

I'm not sure I like the approach of triggering the Jenkins bisection from within the email report logic.
These two steps shouldn't be coupled together.

I think we should create a separate tasks for the Jenkins bisection that is scheduled at the same time when the email report is (and with the same timeout), after all the convoluted logic to handle duplicate email reports in the `SendHandler`.

If we need to shuffle code around to reuse what is being done in the reporting code base, that's fine, but I would prefer to keep the two separate (even if it means creating the same regression data structure twice).

Reply:
Sure, I thought I'd push this first functional version before embarking into what could turn out to be some fairly intrusive refactoring.

Presumably the changes to add extra settings for Jenkins and the added dependency on the `jenkins` module are OK?

Reply:
Yes, the Jenkins config changes are fine and even the refactoring part. We could merge those if you split the PR up into multiple ones.